AK/StringView.h

```diff
@@ -22,6 +22,13 @@ namespace AK {
 class StringView {
 public:
     ALWAYS_INLINE constexpr StringView() = default;
+    template<size_t N>
```

Comment:
Please put newlines around this ctor. I know there aren't newlines around the others, but let's not make an ugly situation uglier.

I think these should also be 2 different commits.
Commit 1 - use StringView in the URL ctors
Commit2 - add the consteval StringView ctor

Reply:
Alternatively, I don't know why we have a `URL(ByteString)` ctor. Seems that can just be removed too - if someone passes a `ByteString` it can be implicitly converted to a `StringView` and hit the `URL(StringView)` ctor.

Reply:
Also let's add a unit test for this ctor